rust/src/lib.rs

```diff
@@ -790,6 +792,62 @@ pub extern "C" fn zcashlc_get_address(
     unwrap_exc_or_null(res)
 }
 
+struct UnifiedAddressParser(UnifiedAddress);
+
+impl zcash_address::TryFromRawAddress for UnifiedAddressParser {
+    type Error = failure::Error;
+
+    fn try_from_raw_unified(
+        data: zcash_address::unified::Address,
+    ) -> Result<Self, zcash_address::ConversionError<Self::Error>> {
+        data.try_into()
+            .map(UnifiedAddressParser)
+            .map_err(|e| format_err!("Invalid Unified Address: {}", e).into())
+    }
+}
+
+/// Returns the transparent receiver within the given Unified Address, if any.
+///
+/// # Safety
+///
+/// - `ua` must be non-null and must point to a null-terminated UTF-8 string containing an
+///   encoded Unified Address.
+/// - Call [`zcashlc_string_free`] to free the memory associated with the returned pointer
+///   when done using it.
+#[no_mangle]
+pub extern "C" fn zcashlc_get_transparent_receiver_for_unified_address(
+    ua: *const c_char,
+) -> *mut c_char {
+    let res = catch_panic(|| {
+        let ua_str = unsafe { CStr::from_ptr(ua).to_str()? };
+
+        let (network, ua) = match ZcashAddress::try_from_encoded(ua_str) {
+            Ok(addr) => addr
+                .convert::<(_, UnifiedAddressParser)>()
+                .map_err(|e| format_err!("Not a Unified Address: {}", e)),
+            Err(e) => return Err(format_err!("Invalid Zcash address: {}", e)),
+        }?;
+
+        if let Some(taddr) = ua.0.transparent() {
+            let taddr = match taddr {
+                TransparentAddress::PublicKey(data) => {
+                    ZcashAddress::from_transparent_p2pkh(network, *data)
+                }
+                TransparentAddress::Script(data) => {
+                    ZcashAddress::from_transparent_p2sh(network, *data)
+                }
+            };
+
+            Ok(CString::new(taddr.encode()).unwrap().into_raw())
+        } else {
+            Err(format_err!(
+                "Unified Address doesn't contain a transparent receiver"
+            ))
```

Comment:
@pacu would you prefer that we return an error if there is no transparent receiver, or just `Ok(null)` (so checking the last error would show nothing, indicating it just didn't have a transparent receiver)? Given that I believe you only plan to call this on UAs generated by the SDK, we can assume a transparent receiver is always intended to be present if this is being called.

Reply:
I think the error is fine.